README.md

```diff
@@ -33,46 +33,63 @@ context after the span finished.
 
 # Example
 
-```rust
-use skywalking::context::trace_context::TracingContext;
-use skywalking::reporter::grpc::Reporter;
-use tokio;
+```rust, no_run
+use skywalking::context::tracer::Tracer;
+use skywalking::reporter::grpc::GrpcReporter;
+use std::error::Error;
+use std::sync::Arc;
+use tokio::signal;
+use tokio::sync::oneshot;
+
+async fn handle_request(tracer: Arc<Tracer<GrpcReporter>>) {
+    let mut ctx = tracer.create_trace_context();
 
-async fn handle_request(reporter: ContextReporter) {
-    let mut ctx = TracingContext::default("svc", "ins");
     {
         // Generate an Entry Span when a request
         // is received. An Entry Span is generated only once per context.
-        let span = ctx.create_entry_span("operation1").unwrap();
+        let span = ctx.create_entry_span("op1").unwrap();
 
         // Something...
 
         {
             // Generates an Exit Span when executing an RPC.
-            let span2 = ctx.create_exit_span("operation2").unwrap();
-            
+            let span2 = ctx.create_exit_span("op2", "remote_peer").unwrap();
+
             // Something...
 
             ctx.finalize_span(span2);
         }
 
         ctx.finalize_span(span);
     }
-    reporter.send(context).await;
+
+    tracer.finalize_context(ctx);
 }
 
 #[tokio::main]
-async fn main() {
-    let tx = Reporter::start("http://0.0.0.0:11800").await;
+async fn main() -> Result<(), Box<dyn Error>> {
+    let reporter = GrpcReporter::connect("http://0.0.0.0:11800").await?;
+    let tracer = Arc::new(Tracer::new("service", "instance", reporter));
+    let (shutdown_tx, shutdown_rx) = oneshot::channel();
+
+    tokio::spawn(handle_request(tracer.clone()));
 
-    // Start server...
+    // Block to report.
```

Comment:
What does block to report work for?

Reply:
> What does block to report work for?

Because dead loop to receive segment to collect, and wait a shutdown signal, will block.

Reply:
How about changing this to

Open reporter as daemon thread(or something more precious in rust) to run.

Block seems a negative word.

WDYT?

Reply:
Already collect as daemon, the block is waiting shutdown signal, otherwise the graceful shutdown cannot be done.